### DISCUSSION TITLE
failed to run custom build command for `llvm-sys v150.1.2

### DISCUSSION THREAD

**Discussion by imedjendoubi** [vosen/ZLUDA#209] [Q&A] [answered]
Any idea how can I solve this?

Ubuntu 22.04.4 LTS

Compiling ascii-canvas v3.0.0
error: failed to run custom build command for `llvm-sys v150.1.2 (/home/preacher/zluda/ext/llvm-sys.rs)`

Caused by:
  process didn't exit successfully: `/home/preacher/zluda/target/release/build/llvm-sys-fe13c4e2d65762e1/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build.rs
  cargo:rerun-if-changed=/home/preacher/zluda/ext/llvm-project/llvm
  cargo:rerun-if-changed=/home/preacher/zluda/ext/llvm-sys.rs/build.cmake
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/home/preacher/zluda/target/release/build/llvm-sys-c3cdaba3a1719214/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/home/preacher/zluda/ext/llvm-project/llvm" "-DLLVM_ENABLE_LIBXML2=OFF" "-DLLVM_ENABLE_ZLIB=OFF" "-DLLVM_ENABLE_ZSTD=OFF" "-DLLVM_ENABLE_CURL=OFF" "-DLLVM_ENABLE_HTTPLIB=OFF" "-DLLVM_ENABLE_LIBEDIT=OFF" "-DLLVM_ENABLE_TERMINFO=OFF" "-DLLVM_BUILD_TOOLS=OFF" "-DLLVM_TARGETS_TO_BUILD=" "-DLLVM_ENABLE_PROJECTS=" "-DCMAKE_PROJECT_INCLUDE_BEFORE=/home/preacher/zluda/ext/llvm-sys.rs/build.cmake" "-DCMAKE_INSTALL_PREFIX=/home/preacher/zluda/target/release/build/llvm-sys-c3cdaba3a1719214/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Release"

  --- stderr
  CMake Error: The source directory "/home/preacher/zluda/ext/llvm-project/llvm" does not exist.
  Specify --help for usage, or press the help button on the CMake GUI.
  thread 'main' panicked at /home/preacher/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...

**Comment by vosen** [comment, marked as answer]
You've probably cloned without git submodules. Use `git clone --recurse-submodules https://github.com/vosen/zluda.git` when cloning or initialize git submodule post-clone

**Comment by NIU-ZZQ** [comment]
maybe you have't intealled cmake
